pubsub/azuresb/azuresb.go

```diff
@@ -496,11 +496,14 @@ func (s *subscription) CanNack() bool {
 	if s == nil {
 		return false
 	}
-	return s.opts.AckFuncForReceiveAndDelete == nil
+	return !s.opts.ReceiveAndDelete
 }
 
 // SendNacks implements driver.Subscription.SendNacks.
 func (s *subscription) SendNacks(ctx context.Context, ids []driver.AckID) error {
+	if !s.CanNack() {
+		panic("unreachable")
```

Comment:
Is this something the user can get to? Now sure "unreachable" is the most informative error message

Reply:
No, the user cannot get here. Since `CanNack` returns `false`, the concrete type should never call `SendNacks`. This is more of an assertion to make the code more clear.